packages/itwinui-react/src/core/DatePicker/DatePicker.tsx

```diff
@@ -285,6 +286,16 @@ export const DatePicker = React.forwardRef((props, forwardedRef) => {
     ...rest
   } = props;
 
+  const logWarning = useWarningLogger();
+
+  const onlyOneRangePropPassed =
+    (!!startDate ? 1 : 0) + (!!endDate ? 1 : 0) === 1;
+  if (enableRangeSelect && onlyOneRangePropPassed) {
```

Comment:
Make sure to add a check for `process.env.NODE_ENV === "development"`, otherwise it will be included in the production bundle.